exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -1500,3 +1503,64 @@ func (ku *Kucoin) ValidateAPICredentials(ctx context.Context, assetType asset.It
 	_, err := ku.UpdateAccountInfo(ctx, assetType)
 	return ku.CheckTransientError(err)
 }
+
+// GetFuturesContractDetails returns details about futures contracts
+func (ku *Kucoin) GetFuturesContractDetails(ctx context.Context, item asset.Item) ([]futures.Contract, error) {
+	if !item.IsFutures() {
+		return nil, futures.ErrNotFuturesAsset
+	}
+	if !ku.SupportsAsset(item) || item != asset.Futures {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, item)
+	}
+
+	contracts, err := ku.GetFuturesOpenContracts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := make([]futures.Contract, len(contracts))
+	for i := range contracts {
+		var cp, underlying currency.Pair
+		underlying, err = currency.NewPairFromStrings(contracts[i].BaseCurrency, contracts[i].QuoteCurrency)
+		if err != nil {
+			return nil, err
+		}
+		cp, err = currency.NewPairFromStrings(contracts[i].BaseCurrency, contracts[i].Symbol[len(contracts[i].BaseCurrency):])
+		if err != nil {
+			return nil, err
+		}
+		settleCurr := currency.NewCode(contracts[i].SettleCurrency)
+		var ct futures.ContractType
+		if contracts[i].ContractType == "FFWCSX" {
+			ct = futures.Perpetual
+		} else {
+			ct = futures.Quarterly
+		}
+		contractSettlementType := futures.Linear
+		if contracts[i].IsInverse {
+			contractSettlementType = futures.Inverse
+		}
+		timeOfCurrentFundingRate := time.Now().Add((time.Duration(contracts[i].NextFundingRateTime) * time.Millisecond) - time.Hour*8).UTC()
```

Comment:
```suggestion
		timeOfCurrentFundingRate := time.Now().Add((time.Duration(contracts[i].NextFundingRateTime) * time.Millisecond) - time.Hour*8).Truncate(time.Hour).UTC()
```

This way it's more precise